ath79: add basic kernel 6.6 support

DragonBluep opened the pull request.

PolynomialDivision commented:
```
drivers/mfd/syscon.c: In function 'of_syscon_register':
drivers/mfd/syscon.c:53:31: error: unused variable 'reset' [-Werror=unused-variable]
   53 |         struct reset_control *reset;
      |                               ^~~~~
cc1: all warnings being treated as errors
```

john-tho commented on target/linux/ath79/patches-6.6/820-mfd-syscon-remove-reset-control-for-syscon-devices.patch:
```diff
@@ -0,0 +1,67 @@
+From: Shiji Yang <yangshiji66@outlook.com>
+Date: Wed, 13 Mar 2024 22:36:31 +0800
+Subject: [PATCH] mfd: syscon: remove reset control for syscon devices
```
Could we consider a different approach to this 6.3 mfd syscon reset issue here? If we one day need to use a driver that relies on this upstream behaviour, we would need to come back to this.
Best wait to see what committers think before changing, I don't want to run you in circles.

Perhaps we:
- add a gpio property (or new `syscon-no-reset`? compatible) to tell syscon not to claim reset?
  ```c
   if (!of_property_read_bool(np, "syscon-no-reset")) {
   	reset = of_reset_control_get_optional_exclusive(np, NULL);
  ```
- work out how we go about having the mfd (sub)device driver (ag71xx here) claim the reset instead?

@dangowrt made some comments upstream, maybe they might have ideas here?

Cheers

DragonBluep replied:
> If we one day need to use a driver that relies on this upstream behaviour, we would need to come back to this.

I believe this will never happen, the ath79 target is already very old.

> * add a gpio property (or new `syscon-no-reset`? compatible) to tell syscon not to claim reset?

A good idea. If more reviewers request a new property here, I'll add it. Thanks.

> * work out how we go about having the mfd (sub)device driver (ag71xx here) claim the reset instead?

At least the `shared` reset controller doesn't work. I've already tried it. I have no idea now.

Anyway, this commit looks strange to me. It didn't export the requested reset controller at all.
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit?id=7d1e3bd94828ad9fc86f55253cd6fec8edd65394

timocapa commented:
Quick run test on a FritzBox 7490’s WASP (https://gitlab.timodo.de/timo/openwrt/-/commits/x490wasp-6.6/?ref_type=heads)

Builds, boots and runs but had one ath10k crash so far. Might be unrelated to 6.6 however as QCA9880 hasn’t been the most reliable historically.

![IMG_0834](https://github.com/openwrt/openwrt/assets/22844613/15e70325-ba13-4f05-808e-4513276840c8)

castillofrancodamian commented:
> Quick run test on a FritzBox 7490’s WASP (https://gitlab.timodo.de/timo/openwrt/-/commits/x490wasp-6.6/?ref_type=heads)
> 
> Builds, boots and runs but had one ath10k crash so far. Might be unrelated to 6.6 however as QCA9880 hasn’t been the most reliable historically.
> 
> ![IMG_0834](https://private-user-images.githubusercontent.com/22844613/312786927-15e70325-ba13-4f05-808e-4513276840c8.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA1MjQzMTgsIm5iZiI6MTcxMDUyNDAxOCwicGF0aCI6Ii8yMjg0NDYxMy8zMTI3ODY5MjctMTVlNzAzMjUtYmExMy00ZjA1LTgwOGUtNDUxMzI3Njg0MGM4LmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxNVQxNzMzMzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yODdiMmVlZmYyNmI4MTA1MzBmOGQyZjRiNTBhMzkwNTYyMGE3YTYwOGM4OTdkMzlmODEyN2U3NDdjNjIwOWQyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.D_mZR-FOZs2t8mkImybsIm_GpqwLh8DbIjc9hRr9J68)

You probably have the same problem as me [here](https://github.com/openwrt/openwrt/issues/14541).

timocapa commented:
@castillofrancodamian No, the 5 GHz radio comes up without issues and I believe hasn’t crashed since either.

Your firmware extraction is broken.

DragonBluep committed:
ath79: skip reset control for syscon devices

Fix the issue of exclusive reset controller acquisition failure on
kernel 6.6.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

DragonBluep committed:
ath79: rename patches directory and kernel config from 6.1 to 6.6

This is preparation for 6.6 kernel support. Renaming files allows
us to better track their git history.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

DragonBluep committed:
ath79: refresh 6.6 kernel patches

Some APIs have been changed. Porting them to the new kernel 6.6
version to fix the compile error.

Removed upstreamed:
	010-v6.4-gpio-ath79-Convert-to-immutable-irq_chip.patch[1]

Manually rebased:
	700-phy-add-ath79-usb-phys.patch
	721-phy-mdio-bitbang-prevent-rescheduling-during-command.patch. ref:[2]

[1] https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-6.6.y&id=b11ce7e48121a02ceedec9f4dfcab4f2bee8f35f
[2] https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-6.6.y&id=002dd3de097c778a74ae9e47e598bea6ad055af0

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

DragonBluep committed:
ath79: restore kernel 6.1 config files and patches

Copy patches and kernel configs from kernel 6.6 to restore the
default 6.1 kernel support files.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

DragonBluep committed:
ath79: add MikroTik suffix to the local GPIO latch driver

The upcoming 6.6 kernel will introduce a new upstream generic
"gpio-latch" driver. It will conflict with the downstream MikroTik
GPIO latch driver. Let's rename it to avoid any potential issues.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

DragonBluep committed:
ath79: refresh 6.6 kernel config files

All kernel config files are refreshed by
`make kernel_oldconfig CONFIG_TARGET={subtarget_target,subtarget}`

"CONFIG_SQUASHFS_DECOMP_SINGLE=y" is manually selected as all ath79
SoCs are single core processors.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

DragonBluep committed:
ath79: enable 6.6 testing kernel

The ath79 target now supports the vanilla 6.6 kernel. Enjoy it!

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

stklcode commented:
Build and run tested on _Ubiquiti airCube AC_ (ath79/generic, AR9342), reabased onto [0c96d20](https://github.com/openwrt/openwrt/commit/0c96d20bf9fba6d814efa88c6fb08a5df094103a) with some additional 6.9 backports and LED patches.

DragonBluep committed:
ath79: fix build error for downstream gpio drivers

Initialize fwnode instead of the deprecated of_node.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

CharlesMengCA commented:
Built and run on Netgear wndr4300v1.  (Ath79/NAND)

Boot: OK.
Modify configuration trough Luci: OK

Since it's a backup router, didn't run for a long time.

skramstad commented:
Tested on:
```
-----------------------------------------------------
 OpenWrt SNAPSHOT, r26128+9-9e86e0b33b
-----------------------------------------------------
system type		: Qualcomm Atheros QCA956X ver 1 rev 0
machine			: Ubiquiti AmpliFi Router HD
```
All good so far. Would be interesting to see if DSA works on 6.6, and doesn't stop after 15-20 ish minutes.

DragonBluep commented:
Changes since 2024.03.16:
1. Fix build error on MikroTik GPIO driver. 
2. Using `syscon-no-reset` property to workaround the reset controller issue instead of removing all related codes.

PolynomialDivision commented:
What is the status of this PR?

DragonBluep commented:
> What is the status of this PR?

It should be ready now. But some existing patches have been updated recently, so I have to manually rebase it.

DragonBluep commented:
> Is it ok to install it on daily use router or are there any known really annoying bugs?

No bug report yet.

> Can I use the current snapshot or should I build it with specific configs ?

Snapshot default kernel is 6.1. You need to build it yourself  by adding `CONFIG_TESTING_KERNEL=y`.

OW87 commented:
Do you have a process with the list of commands so I can build it correctly ?

DragonBluep commented:
The SNAPSHOT kernel will switch to 6.6 tomorrow.

ascendbeing commented:
Looks like the 6.6 buildbot failed today (as of 2hr ago). it was on building images, missing signature files or something.
